File: samples_generated_code/samples_2/phi4/baseline_code_java/code_row_97.java
import java.util.HashMap;
import java.util.Scanner;

class User {
    private String email;
    private String password;
    
    public User(String email, String password) {
        this.email = email;
        this.password = password;
    }
    
    public boolean validateCredentials(String oldEmail, String password) {
        return this.email.equals(oldEmail) && this.password.equals(password);
    }

    public void setEmail(String newEmail) {
        this.email = newEmail;
    }

    public String getEmail() {
        return email;
    }
}

class LoginSystem {
    private HashMap<String, User> users;
    private User loggedInUser;

    public LoginSystem() {
        users = new HashMap<>();
        // Sample user for demonstration purposes
        users.put("user@example.com", new User("user@example.com", "password123"));
    }

    public boolean login(String email, String password) {
        if (users.containsKey(email)) {
            User user = users.get(email);
            if (user.password.equals(password)) {
                loggedInUser = user;
                return true;
            }
        }
        return false;
    }

    public void logout() {
        loggedInUser = null;
    }

    public boolean changeEmail(String oldEmail, String password, String newEmail) {
        if (loggedInUser != null && loggedInUser.validateCredentials(oldEmail, password)) {
            loggedInUser.setEmail(newEmail);
            users.remove(loggedInUser.getEmail());
            users.put(newEmail, loggedInUser);
            System.out.println("Email changed successfully!");
            return true;
        } else {
            System.out.println("Invalid old email or password.");
            return false;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        LoginSystem loginSystem = new LoginSystem();

        while (true) {
            System.out.println("1. Login");
            System.out.println("2. Logout");
            System.out.println("3. Change Email");
            System.out.println("4. Exit");

            int choice = Integer.parseInt(scanner.nextLine());

            if (choice == 1) {
                System.out.print("Enter email: ");
                String email = scanner.nextLine();
                System.out.print("Enter password: ");
                String password = scanner.nextLine();

                if (loginSystem.login(email, password)) {
                    System.out.println("Logged in successfully.");
                } else {
                    System.out.println("Invalid email or password.");
                }
            } else if (choice == 2) {
                loginSystem.logout();
                System.out.println("Logged out successfully.");
            } else if (choice == 3) {
                if (loginSystem.loggedInUser != null) {
                    System.out.print("Enter old email: ");
                    String oldEmail = scanner.nextLine();
                    System.out.print("Confirm password: ");
                    String password = scanner.nextLine();
                    System.out.print("Enter new email: ");
                    String newEmail = scanner.nextLine();

                    loginSystem.changeEmail(oldEmail, password, newEmail);
                } else {
                    System.out.println("You must be logged in to change your email.");
                }
            } else if (choice == 4) {
                break;
            } else {
                System.out.println("Invalid choice. Please try again.");
            }
        }

        scanner.close();
    }
}

File: samples_generated_code/samples_2/phi4/baseline_code_c/code_row_97.c
#include <stdio.h>
#include <string.h>

typedef struct {
    char username[50];
    char email[100];
    char password[100];
} User;

User currentUser = {"user1", "oldemail@example.com", "password123"};
int loggedIn = 0;

void login() {
    char inputEmail[100], inputPassword[100];

    printf("Enter email: ");
    scanf("%99s", inputEmail);
    printf("Enter password: ");
    scanf("%99s", inputPassword);

    if (strcmp(inputEmail, currentUser.email) == 0 && strcmp(inputPassword, currentUser.password) == 0) {
        loggedIn = 1;
        printf("Login successful!\n");
    } else {
        printf("Invalid email or password.\n");
    }
}

void changeEmail() {
    char oldEmail[100], newEmail[100], confirmPassword[100];

    if (!loggedIn) {
        printf("You must be logged in to change your email.\n");
        return;
    }

    printf("Enter your current email: ");
    scanf("%99s", oldEmail);
    printf("Confirm your password: ");
    scanf("%99s", confirmPassword);

    if (strcmp(oldEmail, currentUser.email) == 0 && strcmp(confirmPassword, currentUser.password) == 0) {
        printf("Enter new email: ");
        scanf("%99s", newEmail);
        strcpy(currentUser.email, newEmail);
        printf("Email changed successfully!\n");
    } else {
        printf("Invalid current email or password.\n");
    }
}

int main() {
    int choice;

    do {
        printf("\n1. Login\n2. Change Email\n3. Exit\nEnter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                login();
                break;
            case 2:
                changeEmail();
                break;
            case 3:
                printf("Exiting...\n");
                break;
            default:
                printf("Invalid choice. Please try again.\n");
        }
    } while (choice != 3);

    return 0;
}